py/expression_handlers.py

```diff
@@ -102,11 +102,9 @@ def handle(self, obj, getter):
             return torch.flip(tensor, (dim,))
         result = tensor.detach().clone()
         pivot = tensor.shape[dim] // 2
-        out_slice = (
-            np.s_[:] if d != dim else np.s_[pivot:] for d in range(tensor.ndim)
-        )
-        in_slice = (np.s_[:] if d != dim else np.s_[:pivot] for d in range(tensor.ndim))
-        result[*out_slice] = torch.flip(tensor[*in_slice], dims=(dim,))
+        out_slice = [np.s_[:] if d != dim else np.s_[pivot:] for d in range(tensor.ndim)]
+        in_slice = [np.s_[:] if d != dim else np.s_[:pivot] for d in range(tensor.ndim)]
+        result[tuple(out_slice)] = torch.flip(tensor[tuple(in_slice)], dims=(dim,))
```

Comment:
thanks for the pull. i guess older Python versions don't like that syntax.

you're building two lists and then converting them to tuples. you can just make the tuples directly:
```python
        out_slice = tuple(
            np.s_[:] if d != dim else np.s_[pivot:] for d in range(tensor.ndim)
        )
        in_slice = tuple(np.s_[:] if d != dim else np.s_[:pivot] for d in range(tensor.ndim))
        result[out_slice] = torch.flip(tensor[in_slice], dims=(dim,))
```

so all you need to do is stick a `tuple` before each of the two generator expressions and remove the `*` when indexing using them.

Reply:
@zer0int do you want to update the pull with the suggested change? alternatively, i can just fix it but you won't get credit for the contribution that way.

by the way, your initial approach with 
```python
result[tuple(out_slice)] = torch.flip(tensor[tuple(in_slice)], dims=(dim,))
```

was also just fine, i'd accept either version.

Reply:
Oh, I misinterpreted that as "you want to fix it" - sorry! Thanks for the opportunity; I committed your suggestion (works fine for me / my Python just the same). Cheers!

Reply:
not a problem, i probably wasn't clear enough. thanks for catching the issue! i have Python 3.12 locally so it's easy to accidentally write something that isn't compatible with older versions.